mne/channels/channels.py

```diff
@@ -1536,3 +1539,150 @@ def make_1020_channel_selections(info, midline="z"):
+    if not isinstance(keep_stim, bool):
+        raise TypeError('"keep_stim" must be of type bool, not '
+                        f'{type(keep_stim)}.')
+    if keep_stim is True:
+        try:
+            combined_inst = instance.copy().pick_types(meg=False, stim=True)
+            # Create clean info to avoid inconsistencies
+            stim_ch_names = combined_inst.info['ch_names']
+            stim_ch_types = _get_channel_types(combined_inst.info)
+            combined_inst.info = create_info(sfreq=instance.info['sfreq'],
+                                             ch_names=stim_ch_names,
+                                             ch_types=stim_ch_types)
+        except ValueError:
+            raise ValueError('Could not find stimulus channels.')
+    else:
+        combined_inst = None
+
+    # Get indices of bad channels
+    if not isinstance(drop_bad, bool):
+        raise TypeError('"drop_bad" must be of type bool, not '
+                        f'{type(drop_bad)}.')
+    if drop_bad is False:
+        ch_idx_bad = []
+    elif not instance.info['bads']:
+        ch_idx_bad = []
+    else:
+        ch_idx_bad = pick_channels(instance.info['ch_names'],
+                                   instance.info['bads'])
+
+    # Check correctness of combinations
+    for this_group, these_picks in groups.items():
+        # Check if channel indices are out of bounds
+        if not all(idx in ch_idx for idx in these_picks):
+            raise ValueError('Some channel indices are out of bounds.')
+        # Check if heterogeneous sensor type combinations
+        this_ch_type = np.array(ch_types)[these_picks]
+        if len(set(this_ch_type)) > 1:
+            types = ', '.join(set(this_ch_type))
+            raise ValueError('Cannot combine sensors of different types; '
+                             f'"{this_group}" contains types {types}.')
+        # Remove bad channels
+        these_bads = [idx for idx in these_picks if idx in ch_idx_bad]
+        these_picks = [idx for idx in these_picks if idx not in ch_idx_bad]
+        if these_bads:
+            logger.info('Dropped the following channels in group '
+                        f'{this_group}: {these_bads}')
+        #  Check if combining less than 2 channel
+        if len(set(these_picks)) < 2:
+            warn(f'Less than 2 channels in group "{this_group}"; cannot '
+                 f'combine by method "{method}".')
+        # If all good create more detailed dict without bad channels
+        groups[this_group] = dict(picks=these_picks, ch_type=this_ch_type[0])
+
+    # Combine channels and add them to the new instance
+    group_ch_names, group_ch_types, group_data = [], [], []
+    for this_group, this_group_dict in groups.items():
+        group_ch_names.append(this_group)
+        group_ch_types.append(this_group_dict['ch_type'])
+        these_picks = this_group_dict['picks']
+        this_data = np.take(instance._data, these_picks, axis=ch_axis)
```

Comment:
idem. Don't use private attributes. If its starts with _ it's a private attribute you should not use outside of the class definition.